fastapi_users/db/ormar.py

```diff
@@ -0,0 +1,128 @@
+import datetime
+from typing import Any, List, Optional, Type
+
+import ormar
+from ormar.exceptions import NoMatch
+from pydantic import UUID4
+
+from fastapi_users.db.base import BaseUserDatabase
+from fastapi_users.models import UD, BaseOAuthAccount
+
+
+class OrmarBaseUserModel(ormar.Model):
+    class Meta:
+        tablename = "users"
+        abstract = True
+
+    # sqlalchemy adapter uses char(36) so for ormar it's string format
+    id = ormar.UUID(primary_key=True, uuid_format="string")
+    email = ormar.String(index=True, unique=True, nullable=False, max_length=255)
+    hashed_password = ormar.String(nullable=False, max_length=255)
+    is_active = ormar.Boolean(default=True, nullable=False)
+    is_superuser = ormar.Boolean(default=False, nullable=False)
+    is_verified = ormar.Boolean(default=False, nullable=False)
+
+
+class OrmarBaseOAuthAccountModel(ormar.Model):
+    class Meta:
+        tablename = "oauth_accounts"
+        abstract = True
+
+    id = ormar.UUID(primary_key=True, uuid_format="string")
+    oauth_name = ormar.String(nullable=False, max_length=255)
+    access_token = ormar.String(nullable=False, max_length=255)
+    expires_at = ormar.Integer(nullable=True)
+    refresh_token = ormar.String(nullable=True, max_length=255)
+    account_id = ormar.String(index=True, nullable=False, max_length=255)
+    account_email = ormar.String(nullable=False, max_length=255)
+    # added to keep ordering by creation_date
+    created_date = ormar.DateTime(default=datetime.datetime.now)
```

Comment:
Since some backends (i.e. postgress) do not store the rows in order `ormar` always issues a sort (order_by) even if you do not provide one explicitly. By default it's a primary_key column and since you use `uuids` the order can be pseudo random.

I didn't know if you care to return them in order assumed so by looking at this test that is copied from other db adapters (tortoise etc.).
```python
 # Update
    user_db.oauth_accounts[0].access_token = "NEW_TOKEN"
    await ormar_user_db_oauth.update(user_db)

    # Get by id
    id_user = await ormar_user_db_oauth.get(user.id)
    assert id_user is not None
    assert id_user.id == user_db.id
    # if you do not care about the order here you need to check by id of oauth as you
    # are not guaranteed to get the same oauth in same place without explicit ordering after update
    # since most of the backends (incuding ormar) deletes and recreates the oauths
    # i.e. to populate the default values on model set as server_defaults -> you need to refetch the oauths from db
    # and not just return the object you pass to update
    assert id_user.oauth_accounts[0].access_token == "NEW_TOKEN"
```

If you do not care this column can be removed and the test changed to find oauth by id and not by index in a list.

Reply:
Ok, I see. I can confirm you that the order doesn't matter. The fact that I'm using `[0]` index in the unit test is more a lazy/convenient shortcut than a real feature requirement 😄

You can remove the `created_date` field then ; as for the test, I'll revisit this later.